### PR TITLE
Read CSV file and add a group

### LINKED ISSUE #33: Import from CSV
As a tutor I can import data so that I do not need to manually enter them.

### PR DESCRIPTION
**Pull request by neoruien**
Close #33

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/AddGrpCommand.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+        + PREFIX_GRP + "GRP "
+        + PREFIX_PATH + "PATH ";
+
+    public static final String MESSAGE_SUCCESS = "New tutorial group added: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This tutorial group already exists.";
+
+    private final Group toAdd;
+
+    /**
+     * Creates an AddCommand to add the specified {@code Person}
```
Comment (chunyongg) [inline src/main/java/seedu/address/logic/commands/AddGrpCommand.java]: Javadocs to be updated {@code Person}

File: src/main/java/seedu/address/logic/commands/AddGrpCommand.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        requireNonNull(group);
+        toAdd = group;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+
+        if (model.hasGroup(toAdd)) {
+            throw new CommandException(MESSAGE_DUPLICATE_PERSON);
```
Comment (chunyongg) [inline src/main/java/seedu/address/logic/commands/AddGrpCommand.java]: Same for this, perhaps this is MESSAGE_DUPLICATE_GROUP?

File: src/main/java/seedu/address/logic/parser/AddGrpCommandParser.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        Path path = Paths.get(filePath);
+        CSVUtil csvutil = new CSVUtil(path);
+        Set<Student> students = csvutil.readStudentsFromCsv();
+
+        Group group = new Group(name, students);
+
+        return new AddGrpCommand(group);
+    }
+
+    private static boolean arePrefixesPresent(ArgumentMultimap argumentMultimap, Prefix... prefixes) {
```
Comment (chunyongg) [inline src/main/java/seedu/address/logic/parser/AddGrpCommandParser.java:44]: Is this to check whether PREFIX_PATH is present?

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -20,25 +25,33 @@
...
      */
-    public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs) {
+    public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs,
+        ReadOnlySerenity serenity) {
         super();
-        requireAllNonNull(addressBook, userPrefs);
+        requireAllNonNull(addressBook, userPrefs, serenity);
 
-        logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs);
+        logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs
```
Comment (chunyongg) [inline src/main/java/seedu/address/model/ModelManager.java]: Should this be removed?

File: src/main/java/seedu/address/model/ReadOnlySerenity.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
+package seedu.address.model;
+
+import java.util.List;
+
+import seedu.address.model.serenity.Group;
+
+/**
+ * Unmodifiable view of an address book
```
Comment (chunyongg) [inline src/main/java/seedu/address/model/ReadOnlySerenity.java]: Modify javadocs

File: src/main/java/seedu/address/model/Serenity.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    private final UniqueGroupList groups;
+
+    {
+        groups = new UniqueGroupList();
+    }
+
+    public Serenity() {}
+
+    public Serenity(ReadOnlySerenity toBeCopied) {
+        this();
```
Comment (chunyongg) [inline src/main/java/seedu/address/model/Serenity.java:36]: Are you calling Serenity() ? (empty constructor)

File: src/main/java/seedu/address/model/Serenity.java [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+public class Serenity implements ReadOnlySerenity {
+
+    private final UniqueGroupList groups;
+
+    {
+        groups = new UniqueGroupList();
+    }
+
+    public Serenity() {}
```
Comment (chunyongg) [inline src/main/java/seedu/address/model/Serenity.java:30]: Empty constructor

File: src/main/java/seedu/address/model/group/Group.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+
+    public Set<Student> getStudents() {
+        return Collections.unmodifiableSet(students);
+    }
+
+    public Set<Class> getClasses() {
+        return Collections.unmodifiableSet(classes);
+    }
+
+    public boolean isSameGroup(Group otherGroup) {
```
Comment (chunyongg) [inline src/main/java/seedu/address/model/group/Group.java:82]: This feels very similar to equals()

File: src/main/java/seedu/address/logic/parser/AddGrpCommandParser.java (hunk 3 above)
Comment (neoruien) [reply src/main/java/seedu/address/logic/parser/AddGrpCommandParser.java:44]: this checks whether PREFIX_PATH and PREFIX_GRP are present

File: src/main/java/seedu/address/model/Serenity.java (hunk 6 above)
Comment (neoruien) [reply src/main/java/seedu/address/model/Serenity.java:36]: yes. it's derived from AB3. i believe it is to support the Constructor method at line 21

File: src/main/java/seedu/address/model/Serenity.java (hunk 7 above)
Comment (neoruien) [reply src/main/java/seedu/address/model/Serenity.java:30]: its also derived from AB3. i added some comment there too

File: src/main/java/seedu/address/model/group/Group.java (hunk 8 above)
Comment (neoruien) [reply src/main/java/seedu/address/model/group/Group.java:82]: It is different from equals(). It only checks whether both groups of the same name have _at least one_ other identity field that is the same.

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (neoruien) [reply src/main/java/seedu/address/model/ModelManager.java]: i think it can haha

File: README.md [hunk 9]
```diff
@@ -4,7 +4,3 @@
 <p align="center">
 Serenity is a one-stop software for CS2101 tutors to manage your CS2101 classes.
 </p>
-
-Check out our [User Guide](docs/UserGuide.md) to get started!
```
Comment (chunyongg) [inline README.md:8]: This cannot be deleted (their requirements)

File: README.md [hunk 10]
```diff
@@ -4,7 +4,3 @@
 <p align="center">
 Serenity is a one-stop software for CS2101 tutors to manage your CS2101 classes.
 </p>
-
-Check out our [User Guide](docs/UserGuide.md) to get started! 
-
```
Comment (chunyongg) [inline README.md:9]: This too

**Review by chunyongg — APPROVED** [review]
LGTM!

**Pull request merged**